packages/core/admin/admin/src/pages/ProfilePage/index.js

```diff
@@ -1,160 +1,268 @@
-//                   </ProfilePageLabel>
-//                   <Col xs="6">
-//                     <Select
-//                       aria-labelledby="interface-language"
-//                       selectedValue={get(modifiedData, 'preferedLanguage')}
-//                       onChange={nextLocaleCode => setField('preferedLanguage', nextLocaleCode)}
-//                     >
-//                       {Object.keys(localeNames).map(language => {
-//                         const langName = localeNames[language];
-
-//                         return (
-//                           <Option value={language} key={language}>
-//                             {langName}
-//                           </Option>
-//                         );
-//                       })}
-//                     </Select>
-
-//                     <Padded size="sm" top bottom>
-//                       <Text color="grey">
-//                         {formatMessage({
-//                           id: 'Settings.profile.form.section.experience.interfaceLanguage.hint',
-//                         })}
-//                       </Text>
-//                     </Padded>
-//                   </Col>
-//                 </FormBloc>
-//               </Padded>
-//             </>
-//           )}
-//         </ContainerFluid>
-//       </form>
-//       <BaselineAlignment bottom size="80px" />
-//     </>
-//   );
-// };
-
-// export default ProfilePage;
-
-export default () => 'TODO profile page';
+import React from 'react';
+import {
+  CustomContentLayout,
+  Form,
+  GenericInput,
+  useAppInfos,
+  useFocusWhenNavigate,
+  useNotification,
+  useOverlayBlocker,
+  auth,
+} from '@strapi/helper-plugin';
+import { useIntl } from 'react-intl';
+import { Formik } from 'formik';
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+import pick from 'lodash/pick';
+import omit from 'lodash/omit';
+import { Helmet } from 'react-helmet';
+import { Main } from '@strapi/parts/Main';
+import { H3 } from '@strapi/parts/Text';
+import { Box } from '@strapi/parts/Box';
+import { HeaderLayout } from '@strapi/parts/Layout';
+import { Button } from '@strapi/parts/Button';
+import { Grid, GridItem } from '@strapi/parts/Grid';
+import { Stack } from '@strapi/parts/Stack';
+import { Select, Option } from '@strapi/parts/Select';
+import CheckIcon from '@strapi/icons/CheckIcon';
+import useLocalesProvider from '../../components/LocalesProvider/useLocalesProvider';
+import { fetchUser, putUser } from './utils/api';
+import { schema, layout } from './utils';
+
+const ProfilePage = () => {
+  const { changeLocale, localeNames } = useLocalesProvider();
+  const { setUserDisplayName } = useAppInfos();
+  const queryClient = useQueryClient();
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { lockApp, unlockApp } = useOverlayBlocker();
+  useFocusWhenNavigate();
+
+  const { status, data } = useQuery('user', () => fetchUser(), {
```

Comment:
Can you pass the notifyStatus function returned by the useNotifyAT hook in order to notify the user that the data has been loaded?